ports/stm32/dac.c

```diff
@@ -485,6 +485,11 @@ mp_obj_t pyb_dac_write_timed(size_t n_args, const mp_obj_t *pos_args, mp_map_t *
     #endif
     }
 
+    #if defined(STM32F7) || defined(STM32H7)
```

Comment:
You should be able to remove this if conditional.  The `MP_HAL_CLEAN_DCACHE` macro is defined to nothing on MCUs that don't have a D-cache.